projects/storefrontlib/cms-structure/guards/cms-page-guard.service.ts

```diff
@@ -71,11 +69,6 @@ export class CmsPageGuardService {
     state: RouterStateSnapshot
   ): Observable<boolean | UrlTree> {
     return this.cmsService.getPageComponentTypes(pageContext).pipe(
-      map((componentTypes) =>
-        pageContext.id === SMART_EDIT_CONTEXT
-          ? [SMART_EDIT_DUMMY_COMPONENT_TYPE, ...componentTypes]
-          : componentTypes
-      ),
```

Comment:
[NITPICK]
Theoretically, removing the `const SMART_EDIT_DUMMY_COMPONENT_TYPE` from the public API and removing this behavior of loading `SmartEditModule` when the cms component type `SMART_EDIT_DUMMY_COMPONENT_TYPE` is present - those are breaking changes.

However, we need to observe that:
- the probability of customers using `SMART_EDIT_DUMMY_COMPONENT_TYPE` is very low
- the probability they dependend on the behavior that `SMART_EDIT_DUMMY_COMPONENT_TYPE` triggered the lazy-loading of `SmartEditModule` is very low
- by introducing `this.featureModules?.resolveFeature(SMART_EDIT_FEATURE).subscribe();` to the `SmartEditLauncherService` we preserve the original behavior - when the route is `/cx-preview`, the `SmartEditModule` is being lazy loaded. But now the logic is much simpler without unnecessary workarounds and depending on fake CMS component types and CmsGuard.

All that said, I agree to turn a blind eye to this slight breaking change.